main.cpp

```diff
@@ -0,0 +1,184 @@
+                {
+                    if(animals.size() != 0)
+                    {
+                        for(int i = 0; i < animals.size(); i ++)
+                        cout << "Nume: " << animals[i].Nume << " " << "Specie: " << animals[i].Specie << " " << "Categorie de productie: " << animals[i].CategorieDeProductie
+                        << " " << "Greutate: " << animals[i].Greutate << " " << "Varsta: " << animals[i].Varsta << endl;
+                    }
+                    else
+                        cout << "Niciun animal intordus!" << endl;
+
+            break;
+                }
+
+            case 3: // Modifica detaliile unui animal
+                {
+                     if(animals.size() != 0)
+                    {
+                        cout << "Introduceti numele animalului pentru care doriti sa faceti modificarea: ";
+                        cin >> animalDeCautat;
+                        auto it = findAnimalByName(animals, animalDeCautat);//auto it = find(animals.begin(), animals.end(), animalDeCautat); // cauta in vector dupa nume si pozitie
+                        if (it != animals.end())
+                        {
+                            int index = distance(animals.begin(), it); // calculeaza pozitia in functie de distanta de la inceputul vectorului pana la iterator
+                            string nume, specie, rasa, categoriedeproductie;
+                            int varsta, greutate;
+                            cout << "Introduceti modificarile: ";
+                            cout << "Nume: ";
+                            cin >> nume;
+                            cout << "Specie: ";
+                            cin >> specie;
+                            cout << "Rasa: ";
+                            cin >> rasa;
+                            cout << "Categorie de productie: ";
+                            cin >> categoriedeproductie;
+                            cout << "Greutate: ";
+                            cin >> greutate;
+                            cout << "Varsta: ";
+                            cin >> varsta;
+
+                            *it = Animal(nume, specie, rasa, categoriedeproductie, greutate, varsta);
+
+                        }
+                        else
+                            cout << "Animalul nu exista!" << endl;
+                    }
+
+                    else
+                        cout << "Niciun animal intordus!" << endl;
+
+                }
+            break;
+
+            case 4: //  sterge un element din lista
+                {
+                    if(animals.size() != 0)
+                    {
+                        cout << "Introduceti numele animalului pe care doriti sa il stergeti: ";
+                        cin >> animalDeSters;
+                        auto itToDelete = findAnimalByName(animals, animalDeSters);
+                        if (itToDelete != animals.end())
+                            {
```

Comment:
aici esti sigura ca nu trebuie `>` in loc de `!=`?

Reply:
Da pai, trebuie sa verifice daca a ajunsd la sfarsitul vectorului sau nu. De ce sa fie '>' ?